### ISSUE TITLE
Window will not stay shown on launch

### ISSUE THREAD

**Issue by cwhits** [maxogden/menubar#76]
If I run the app from the command line with electron `npm start` I can have the menubar window start off as shown using either `mb.window.show()` or `mb.showWindow()`.

However, once I package the app with electron-packager and launch the .app, the window flashes for a brief moment but won't stay shown.

electron 0.33.1, menubar 2.2.1

```javascript
var menubar = require('menubar')

var mb = menubar()

mb.on('ready', function ready () {
  console.log('app is ready')
  mb.window.show()
})
```

**Comment by maxogden** [comment]
Hmm I have no idea why this is happening, sorry. Will leave it open in case someone else has ideas

**Comment by cwhits** [comment]
In a fun turn of events, it works on Windows (8 at least).

**Comment by maxogden** [comment]
this is most likely an electron issue, I would try to reproduce the issue
using plain electron (e.g. no require('menubar')) and open an issue there
to get it fixed

**Comment by cwhits** [comment]
Plain electron appears to work. It stays shown once built with electron-packager :confused: 
```javascript 
var app = require('app');
var BrowserWindow = require('browser-window');

app.on('ready', function() {
var win = new BrowserWindow({ width: 800, height: 600, show: false });
win.on('closed', function() {
      win = null;
});

win.loadUrl('https://github.com');
win.show();
});
```

**Comment by maxogden** [comment]
@cwhits oh try this menubar code:

```js
var menubar = require('menubar')

var mb = menubar({preloadWindow: true})

mb.on('ready', function ready () {
  console.log('app is ready')
  mb.on('create-window', function () {
    mb.window.show()
  })
})
```

**Comment by cwhits** [comment]
The only other info I have to add is that `menubar.window` thinks that it's visible. I added some logging to the menubar module's index.js file and the value of `menubar.window.isVisible()` is true right after the 'ready' emitter even though I can't see the window.

**Comment by cwhits** [comment]
@maxogden That code doesn't help. `Create-window` doesn't seem to be emitted after `ready`. If I add a `console.log('create-window emitted')` above `mb.window.show()`, it never gets outputted.

**Comment by maxogden** [comment]
alright. I'm on my phone now, but you might be able to figure it out by
looking at the menubar source code

On Wednesday, October 28, 2015, cwhits <notifications@github.com> wrote:

> @maxogden <https://github.com/maxogden> That code doesn't help.
> Create-window doesn't seem to be emitted after ready. If I add a console.log('create-window
> emitted') above mb.window.show(), it never gets outputted.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/maxogden/menubar/issues/76#issuecomment-151968424>.
>


-- 
Sent from my phone

**Comment by cwhits** [comment]
@maxogden I've added logs at almost every point in the menubar source and still couldn't figure it out. Everything appears to be doing what it's supposed to.

I've finally found a workaround but I don't understand why it works.

The screen won't stay displayed if I do:
```javascript
    mb.on('ready', function ready () {
        mb.window.show()
    })
```

**BUT** if I add a 1ms timeout, it stays displayed...

```javascript
    mb.on('ready', function ready () {
        setTimeout(function(){mb.window.show();}, 1);
    })
```

**Comment by jenslind** [comment]
Try show the window on the `after-create-window` event.

```javascript
var menubar = require('menubar')

var mb = menubar({preloadWindow: true})

mb.on('ready', function ready () {
  console.log('ready')
})

mb.on('after-create-window', function () {
  mb.window.show()
})

```

**Comment by cwhits** [comment]
@jenslind Nope. It flashes briefly but doesn't stay open. Doing the 1ms timeout also works as a workaround there too.

**Comment by jenslind** [comment]
After some investigation it looks like `app.dock.hide()` is causing this.
Removing it or moving it to after the window is created worked.

**Comment by vernondegoede** [comment]
This is indeed the solution to this issue @jenslind. Should I make a PR for this?

**Comment by JoeDawson** [comment]
Did anyone manage to solve this?
